python/ray/air/_internal/torch_utils.py

```diff
@@ -481,26 +504,37 @@ def move_tensors_to_device(
         device: The device to move tensors to. If None, tensors are not moved.
         non_blocking: If True, perform device transfer without forcing a
             synchronization.
+        pin_memory: Whether to pin the memory of the tensors. Defaults to False.
 
     Returns:
         The input tensors moved to the specified device
     """
     if device is None:
         return batch
 
+    def to_device(tensor: torch.Tensor) -> torch.Tensor:
+        tensor = tensor_pin_memory_if_needed(tensor, pin_memory)
+
+        return tensor.to(
+            device, non_blocking=is_tensor_non_blocking_transfer(tensor, non_blocking)
+        )
+
     if _is_tensor(batch):
-        return batch.to(device, non_blocking=non_blocking)
+        return to_device(batch)
     elif _is_tensor_sequence(batch):
-        return type(batch)([t.to(device, non_blocking=non_blocking) for t in batch])
+        return type(batch)([to_device(t) for t in batch])
     elif _is_nested_tensor_sequence(batch):
         return type(batch)(
-            [concat_tensors_to_device(t, device, non_blocking) for t in batch]
+            [
+                concat_tensors_to_device(t, device, non_blocking, pin_memory)
+                for t in batch
+            ]
         )
     elif _is_tensor_mapping(batch):
-        return {k: t.to(device, non_blocking=non_blocking) for k, t in batch.items()}
+        return {k: to_device(t) for k, t in batch.items()}
```

Comment:
Refer to tensordict implementation, which performs the pin_memory for multiple with a threadpool so that multiple keys / elements can be pinned concurrently.